### PR TITLE
Fixes the "Extract failed" messages in UnRAR extractor plugin

### PR DESCRIPTION
**Pull request by Arno-Nymous**
See [Issue 2289](https://github.com/pyload/pyload/issues/2289).

### CODE REVIEW

File: module/database/FileDatabase.py [hunk 1]
```diff
@@ -831,7 +832,9 @@ def getFile(self, id):
         self.c.execute("SELECT url, name, size, status, error, plugin, package, linkorder FROM links WHERE id=?", (str(id), ))
         r = self.c.fetchone()
         if not r: return None
-        return PyFile(self.manager, id, * r)
+        pyfile = PyFile(self.manager, id, * r)
```
Comment (Arno-Nymous) [inline module/database/FileDatabase.py]: Admittedly have to mention that I don't know whether this is the right place to re-init the `pyfile.plugin`.
Maybe there is a better place to put in order to remain DRY. But it's a fact that in some place there has to be made sure that `pyfile.plugin` should never be `None` but set to the appropriate plugin.

**Comment by vuolter** [comment]
I'll have a look at this soon I can.

File: module/plugins/hooks/ExtractArchive.py [hunk 2]
```diff
@@ -275,17 +275,17 @@ def extract(self, ids, thread=None):  #@TODO: Use pypack, not pid to improve met
             if not exists(out):
                 os.makedirs(out)
 
-            matched   = False
-            success   = True
-            files_ids = dict((fdata['name'], ((fsjoin(dl_folder, pypack.folder, fdata['name'])), fid, out)) for fid, fdata \
-                        in sorted(pypack.getChildren().values(), key=lambda k: k['name'])).items()  #: Remove duplicates
+            matched = False
+            success = True
+            files_ids = dict((fdata['name'], (fdata['id'], fsjoin(dl_folder, pypack.folder, fdata['name']), out)) for fdata \
```
Comment (Arno-Nymous) [inline module/plugins/hooks/ExtractArchive.py]: Fixes [#2296](https://github.com/pyload/pyload/issues/2296)

**Comment by GammaC0de** [comment]
@Arno-Nymous Maybe your PR does fix the above issue, but it also changes fundamental parts of `pyLoad`'s core so I'm pretty sure that it will be rejected by @vuolter, never the less that it **was** working before (some changes were made to it that broke it) without the changes in the core.. :wink:

**Comment by nordic-style** [comment]
@GammaC0de i did not check the pr but i would say that there are some core changes that are required to "fix" pyload in a proper way?! ;)

**Comment by manuelm** [comment]
The PR is incomplete:
* The order of the list elements in https://github.com/pyload/pyload/pull/2290/files#diff-5bc8c590ca33f5f573f2a9570649a67dR348 needs to be adapted as well.
* `pyfile.plugin` gets released in https://github.com/pyload/pyload/pull/2290/files#diff-5bc8c590ca33f5f573f2a9570649a67dR326 but needs to stay alive until https://github.com/pyload/pyload/pull/2290/files#diff-5bc8c590ca33f5f573f2a9570649a67dR350

**Pull request closed without merging**